core/block/editor/file/file.go

```diff
@@ -220,8 +222,10 @@ func (sf *sfile) updateFile(ctx session.Context, id, groupId string, apply func(
 }
 
 func (sf *sfile) DropFiles(req pb.RpcFileDropRequest) (err error) {
-	if err = sf.Restrictions().Object.Check(model.Restrictions_Blocks); err != nil {
-		return err
+	if layout, ok := sf.Layout(); !ok || layout != model.ObjectType_collection {
```

Comment:
please make a method called isCollection

Reply:
You can make it like so func isCollection(sb any), here you can check everything, is object a smartblock? does it have layout? is the layout collection?